sympy/core/tests/test_function.py

```diff
@@ -1324,6 +1324,31 @@ def test_issue_15084_13166():
     assert eq.diff(x, g(x)) == eq.diff(g(x), x)
 
 
+def test_issue_22439():
+    x, y, h = symbols('x y h')
+    u = Function('u')
+    _xi_1, _xi_2 = [Dummy() for i in range(2)]
+    assert u(x - h, y - h).diff(h, 2) == Subs(Derivative(u(_xi_1, -h + y), (_xi_1, 2)), _xi_1, -h + x) + \
```

Comment:
Reviewing these kind of tests is rather involved since it requires to check the actual result (as in, should the derivative really be equal to what is stated here). Could you create tests based on some prior result from a reliable source (e.g. definitions in wiki pages or wolfram). With a link, this would also make it easier in the future to review if some problem comes up. If it doesn't exist then so be it.

Reply:
Cool , yeah maybe I could try this too. I was quick to realize that the rather difficult part of this PR was to add tests , cause the expression/problem mentioned in the issue reported in Op was not a trivial day to day result which a general sympy user might end up trying ! 
Hence the best I could do was to create pretty printed results of all these tests and add them into a comment for reference which is here https://github.com/sympy/sympy/pull/22806#issuecomment-1007272066